napari/_qt/dialogs/qt_notification.py

```diff
@@ -80,20 +81,18 @@ def __init__(
         """[summary]"""
         super().__init__(None)
 
-        # FIXME: this does not work with multiple viewers.
         # we need a way to detect the viewer in which the error occured.
-        for wdg in QApplication.topLevelWidgets():
-            if isinstance(wdg, QMainWindow):
-                try:
-                    # TODO: making the canvas the parent makes it easier to
-                    # move/resize, but also means that the notification can get
-                    # clipped on the left if the canvas is too small.
-                    canvas = wdg.centralWidget().children()[1].canvas.native
-                    self.setParent(canvas)
-                    canvas.resized.connect(self.move_to_bottom_right)
-                    break
-                except Exception:
-                    pass
+        viewer_instance = get_viewer_instance()
+        try:
+            # TODO: making the canvas the parent makes it easier to
+            # move/resize, but also means that the notification can get
+            # clipped on the left if the canvas is too small.
+            canvas = viewer_instance.canvas.native
```

Comment:
careful when you merge #2611  as it will conflict here

Reply:
I've merged this now. On that note... we probably should raise an error in `get_viewer_instance()` if we don't find one? I know we wanted to get rid of that function soon if possible. Are we going to leave as is for this PR and fix elsewhere?